packages/channel-provider/src/messaging-service.ts

```diff
@@ -109,8 +109,6 @@ export class MessagingService {
         } else if (isJsonRpcErrorResponse(event.data)) {
           reject(new RpcError(event.data.error));
         }
-      } else {
-        logger.error({event}, 'Received a non JsonRpc response');
```

Comment:
The infinite loop here is:
- when the channel provider triggers a log event, it calls `postMessage` to make web3torrent aware of that log event
- but L113 here will read _that_ message, containing the log event, see that it's not a JsonRpc message, and log an error
- logging that error calls `postMessage` ...

## Alternative 1
1. web3torrent listens for log events only from the channel provider
2. wallet logger calls postMessage with a notification that there was a log event
3. channel-provider reads that event, forwards it to web3torrent

## Alternative 2
0. Add a `LogEvent` notification type to the client-wallet-API
1. channel-provider stores the log events and exposes the `saveLogs` function on the window
   a. channel-provider can expose a window function `addToLogs`, or listen to some event from the app, about new log events
2. web3torrent does whatever's done in `a` when it gets a log event
3. channel-provider stores a log event when it receives a `LogEvent` JsonRpc notification

I'm starting to lean towards 2, though I'd prefer to do either of these in a separate PR (and maybe not urgently?)